src/writer/to_code.rs

```diff
@@ -128,17 +128,33 @@ impl ToCode for Chunk {
                 ref signal,
                 ref trampoline,
                 in_trait,
+                is_detailed,
             } => {
-                let s1 = format!(
-                    "connect_raw(self.as_ptr() as *mut _, b\"{}\\0\".as_ptr() as *const _,",
-                    signal
-                );
+                let mut v: Vec<String> = Vec::with_capacity(4);
+                if is_detailed {
+                    v.push(format!(
+                        r#"let detailed_signal_name = detail.map(|name| {{ format!("{0}::{{}}\0", name) }});"#,
+                        signal
+                    ));
+                    v.push(format!(
+                        r#"let signal_name_ptr: *const u8 = (if let Some(ref name) = detailed_signal_name {{ name.as_bytes() }} else {{ &b"{0}\0"[..] }}).as_ptr();"#,
+                        signal
+                    ));
+                }
+                v.push(if !is_detailed {
+                    format!(
+                        "connect_raw(self.as_ptr() as *mut _, b\"{}\\0\".as_ptr() as *const _,",
+                        signal
+                    )
+                } else {
+                    "connect_raw(self.as_ptr() as *mut _, signal_name_ptr as *const _,".to_string()
+                });
```

Comment:
Perhaps you could just merge this with the `if` block above, and add an `else` for it?

```suggestion
                    v.push("connect_raw(self.as_ptr() as *mut _, signal_name_ptr.as_ptr() as *const _,".to_string());
                } else {
                    v.push(format!(
                        "connect_raw(self.as_ptr() as *mut _, b\"{}\\0\".as_ptr() as *const _,",
                        signal
                    ));
                }
```

Reply:
That makes sense. I'll unify those two conditionals.

Reply:
:+1: Sorry for CI noise, this diff already included the move of `.as_ptr()` to the call site, which requires the other diff to be applied as well.

Reply:
No prob, will make sure all's good in the next commit.